askomics/libaskomics/SourceFileConvertor.py

```diff
@@ -66,55 +81,49 @@ def get_first_lines(self, limit):
             if match:
                 continue
 
-            val_cell_type = {}
+            val_cell_type = {} # :: Column Index -> Set Values
             with open(curr_file, 'r') as src:
-                cpt = 0
                 content = []
-                for line in src:
+                for cpt, line in enumerate(src):
+                    cells = line.rstrip('\r\n').split()
                     if cpt == 1:
                         cell_types = []
                         col_cpt = 0
-                        for cell in line.rstrip('\r\n').split():
+                        for cell in cells:
                             if col_cpt == 0:
                                 curr_cell_type = "Entity"
-                            elif cell.isdigit() or is_float(cell):
+                            elif is_decimal(cell):
                                 curr_cell_type = "Numeric"
                             else:
                                 curr_cell_type = "Text"
                                 #OFI : We try to guess if the present colonne is a 'Category'
-                                val_cell_type[col_cpt] = []
-                                val_cell_type[col_cpt].append(cell)
+                                val_cell_type.setdefault(col_cpt, set()).add(cell)
```

Comment:
The purpose of set() is to count the unique values in the column, and compare this number to the number of rows. 
The old code code was building a list, allowing to count the occurrences of each values afterward (a Counter would be more appropriate), but only using the number of unique values.